File: go/routes/rating.routes.go
package routes

import (
	createRating "themoviebakery/controllers/rating-controllers/create"
	getRating "themoviebakery/controllers/rating-controllers/get"
	getRatingHandler "themoviebakery/handlers/rating-handlers/get"

	"github.com/gin-gonic/gin"
)

func InitRatingRoutes(router *gin.Engine) {
	groupRoute := router.Group("/api/v1")
	groupRoute.POST("/rating", createRating.CreateRating)
	groupRoute.GET("/rating", getRating.AllRatingsAllUsers)
	groupRoute.GET("/rating-by-id", getRatingHandler.GetRatingHandler)
	groupRoute.GET("/rating-by-user", getRatingHandler.GetRatingsByUser)
	groupRoute.GET("/rating-user-movie", getRatingHandler.GetRatingUserMovie)
}

File: src/routes/rating.routes.ts
import { Router } from 'express'
import RatingController from '../controllers/rating'

const ratingRouter = Router()
const ratingController = new RatingController()

ratingRouter.post('/create', ratingController.createAndUpdate)
ratingRouter.get('/get-all', ratingController.allRatingsAllUsers)
ratingRouter.get('/get-by-user', ratingController.findByUser)

export default ratingRouter
